src/encoding/eucjp_encoding_object.cpp

```diff
@@ -0,0 +1,82 @@
+#include "natalie.hpp"
+
+namespace Natalie {
+
+// NATFIXME: actual implementation is wrong for much of this file
```

Comment:
Is this comment still accurate, or just leftover while you were working on the file?

If it's still accurate, I hesitate to merge a known-wrong implementation. I worry we'll have subtle bugs that are hard to track down later.

If it's known to be wrong, perhaps we should put a `NAT_NOT_YET_IMPLEMENTED()` in the methods that are wrong... 🤔

Reply:
The comment is still accurate: With the exception of `next_char`, I don't think any of these methods (e.g. `prev_char` or `*_codepoint` related methods) are tested by any specs we have (and possibly not by any specs in ruby's spec).  To that end, these may or may not work, but would venture that the Japanese encodings would give incorrect results for some inputs for some of the methods.  IIRC few if any of these are exposed as ruby methods so that makes it harder to test.

`NAT_NOT_YET_IMPLEMENTED()` is a great idea.  Looks like after adding to N.N.Y.I. the tests still pass which confirms my  suspicion that none of the tests hit that code.

If you'd rather - I can mark as draft, then work on fixing the rest of the methods (and add new tests) before trying to merge.

Reply:
Wait, I think @andrykonchin was adding some sort of test in the `test/` directory for prev_char(). I'm on mobile right now so it's hard to search. But I can check later!

Reply:
OK, I found it here: https://github.com/natalie-lang/natalie/blob/355deeec90ecf5dbd7ac29e8b934c3b76845354e/test/natalie/string_test.rb#L571

For this PR, I'm fine with us using `NAT_NOT_YET_IMPLEMENTED()`, and we can come back to work on those methods in a later PR.

Thanks!